src/main/java/uk/gov/hmcts/ccd/domain/service/createcase/SubmitCaseTransaction.java

```diff
@@ -121,6 +130,12 @@ public CaseDetails submitCase(Event event,
             caseDetailsAfterCallback
         );
 
+        if (this.applicationParams.getCaseGroupAccessFilteringEnabled()) {
+            CaseAccessGroupUtils caseGroupAccessTypeUtils = new CaseAccessGroupUtils();
+            caseGroupAccessTypeUtils.updateCaseAccessGroupsInCaseDetails(caseDetailsAfterCallbackWithoutHashes,
+                caseTypeDefinition, caseDataService);
+        }
```

Comment:
Can we add this same block to be in `CreateCaseEventService` too please? This class only deals with the case creation endpoints, but we need exactly the same for all events.

Reply:
fixed in #2389